File: 4.8-bc/go/stage-1-flat/app/model/order.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Order struct {
	ID         uuid.UUID
	UserID     uuid.UUID
	TotalCents int64
	Status     string
	CreatedAt  time.Time
}

type OrderItem struct {
	ProductID uuid.UUID
	Quantity  int
}

File: 4.8-bc/ts/stage-1-flat/src/service/order.ts
import { randomUUID } from "node:crypto";

import type { Order } from "../model/order.js";
import type { OrderRepository } from "../repository/order.js";

export class OrderService {
  constructor(private readonly repo: OrderRepository) {}

  async place(userId: string, totalCents: number): Promise<Order> {
    const o: Order = {
      id: randomUUID(),
      userId,
      totalCents,
      status: "pending",
      createdAt: new Date(),
    };
    await this.repo.create(o);
    return o;
  }
}
